src/NLog/Config/LoggingRuleLevelFilter.cs

```diff
@@ -0,0 +1,251 @@
+                        _activeFilter = activeFilter = new KeyValuePair<string, bool[]>(levelFilter, logLevels);
+                    }
+                    catch (ArgumentException ex)
+                    {
+                        InternalLogger.Warn(ex, "Logging rule {0} with filter `{1}` has invalid level filter: {2}", _loggingRule.RuleName, _loggingRule.LoggerNamePattern, levelFilter);
+                        return LoggingRuleLevelFilter.Off.LogLevels;
+                    }
+                }
+            }
+
+            return activeFilter.Value;
+        }
+    }
+
+    /// <summary>
+    /// Dynamic filtering with a minlevel and maxlevel range
+    /// </summary>
+    class DynamicRangeLevelFilter : ILoggingRuleLevelFilter
+    {
+        readonly LoggingRule _loggingRule;
+        readonly SimpleLayout _minLevel;
+        readonly SimpleLayout _maxLevel;
+        private KeyValuePair<MinMaxLevels, bool[]> _activeFilter;
+
+        public bool[] LogLevels => GenerateLogLevels();
+
+        public DynamicRangeLevelFilter(LoggingRule loggingRule, SimpleLayout minLevel, SimpleLayout maxLevel)
+        {
+            _loggingRule = loggingRule;
+            _minLevel = minLevel;
+            _maxLevel = maxLevel;
+            _activeFilter = new KeyValuePair<MinMaxLevels, bool[]>(new MinMaxLevels(string.Empty, string.Empty), LoggingRuleLevelFilter.Off.LogLevels);
+        }
+
+        public ILoggingRuleLevelFilter SetLoggingLevels(LogLevel minLevel, LogLevel maxLevel, bool enable)
+        {
+            return new LoggingRuleLevelFilter(LogLevels).SetLoggingLevels(minLevel, maxLevel, enable);
+        }
+
+        private bool[] GenerateLogLevels()
+        {
+            var minLevelFilter = _minLevel?.Render(LogEventInfo.CreateNullEvent()) ?? string.Empty;
+            var maxLevelFilter = _maxLevel?.Render(LogEventInfo.CreateNullEvent()) ?? string.Empty;
+            if (string.IsNullOrEmpty(minLevelFilter) && string.IsNullOrEmpty(maxLevelFilter))
+                return LoggingRuleLevelFilter.Off.LogLevels;
+
+            var activeFilter = _activeFilter;
+            if (!activeFilter.Key.Equals(new MinMaxLevels(minLevelFilter, maxLevelFilter)))
+            {
+                LogLevel minLevel = ParseLogLevel(minLevelFilter, LogLevel.MinLevel) ?? LogLevel.MaxLevel;
+                LogLevel maxLevel = ParseLogLevel(maxLevelFilter, LogLevel.MaxLevel) ?? LogLevel.MinLevel;
+
+                bool[] logLevels = new bool[LogLevel.MaxLevel.Ordinal + 1];
+                for (int i = minLevel.Ordinal; i <= Math.Min(logLevels.Length - 1, maxLevel.Ordinal); ++i)
+                {
+                    logLevels[i] = true;
+                }
+                _activeFilter = activeFilter = new KeyValuePair<MinMaxLevels, bool[]>(new MinMaxLevels(minLevelFilter, maxLevelFilter), logLevels);
+            }
+            return activeFilter.Value;
+        }
+
+        LogLevel ParseLogLevel(string logLevel, LogLevel defaultLevel)
```

Comment:
as there is an argument defaultLevel here, I think it would be logical to have also levelOnError here. Maybe rename defaultLevel to levelIfEmpty.

(defaults are sometimes also used for errors)

Reply:
Fixed